service/create/service.go

```diff
@@ -540,22 +540,40 @@ func (s *Service) addFunc(obj interface{}) {
 		return
 	}
 
-	// Create policy
 	bucketName := s.bucketName(cluster)
 
-	var policy resources.NamedResource
-	var policyErr error
+	// Create master IAM policy
+	var masterPolicy resources.NamedResource
+	var masterPolicyErr error
 	{
-		policy = &awsresources.Policy{
-			ClusterID: cluster.Spec.Cluster.Cluster.ID,
-			KMSKeyArn: kmsKey.Arn(),
-			S3Bucket:  bucketName,
-			AWSEntity: awsresources.AWSEntity{Clients: clients},
+		masterPolicy = &awsresources.Policy{
+			ClusterID:  cluster.Spec.Cluster.Cluster.ID,
```

Comment:
Good point I've seen the getKeys methods in the other operators. There is much refactoring to be done I'll keep this in mind.